dbt_semantic_interfaces/dbt_semantic_interfaces/objects/elements/entity.py

```diff
@@ -0,0 +1,75 @@
+from __future__ import annotations
+
+from typing import Optional, List
+
+from dbt_semantic_interfaces.objects.base import (
+    HashableBaseModel,
+    ModelWithMetadataParsing,
+)
+from dbt_semantic_interfaces.objects.common import Metadata
+from dbt_semantic_interfaces.enum_extension import ExtendedEnum
+from dbt_semantic_interfaces.references import (
+    EntityReference,
+    CompositeSubEntityReference,
+)
+
+
+class EntityType(ExtendedEnum):
+    """Defines uniqueness and the extent to which an entity represents the common entity for a data source"""
+
+    FOREIGN = "foreign"
+    NATURAL = "natural"
+    PRIMARY = "primary"
+    UNIQUE = "unique"
+
+
+class CompositeSubEntity(HashableBaseModel):
+    """CompositeSubEntities either describe or reference the entities that comprise a composite entity"""
+
+    name: Optional[str]
+    expr: Optional[str]
+    ref: Optional[str]
+
+    @property
+    def reference(self) -> CompositeSubEntityReference:  # noqa: D
+        assert (
+            self.name
+        ), f"The element name should have been set during model transformation. Got {self}"
+        return CompositeSubEntityReference(element_name=self.name)
+
+
+class Entity(HashableBaseModel, ModelWithMetadataParsing):
+    """Describes a entity"""
+
+    name: str
+    description: Optional[str]
+    type: EntityType
+    role: Optional[str]
+    entities: List[CompositeSubEntity] = []
+    expr: Optional[str] = None
+    metadata: Optional[Metadata] = None
```

Comment:
We had it as `Optional[Metadata]` without a default value. I added the default of `None` to allow not having to explicitly set it to `None` if we don't pass this property in.